### PR TITLE
Add Structure module (json, csv)

### PR DESCRIPTION
**Pull request by alisedighmoghadam**
This pull request introduces the new Structure Module, designed to address the requirements and objectives outlined in Issue #246. 

@cieslarmichal please review the changes and provide your insights or approval.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -4,3 +4,6 @@
 [submodule "externals/fmt"]
 	path = externals/fmt
 	url = https://github.com/fmtlib/fmt.git
+[submodule "faker-cxx"]
```
Comment (cieslarmichal) [inline .gitmodules:7]: why did you added library to its own dependencies? probably some mistake, please remove

File: include/faker-cxx/Structure.h [hunk 2]
```diff
@@ -0,0 +1,662 @@
...
+#include "Sport.h"
+#include "String.h"
+#include "System.h"
+#include "VideoGame.h"
+#include "Weather.h"
+#include "Word.h"
+
+namespace faker
+{
+struct test
```
Comment (cieslarmichal) [inline include/faker-cxx/Structure.h:40]: test should not be in api header file, remove please

File: include/faker-cxx/Structure.h [hunk 3]
```diff
@@ -0,0 +1,662 @@
...
+    WordAdverb,
+    WordConjuction,
+    WordInterjection,
+    WordNoun,
+    WordPreposition,
+    WordVerb,
+
+};
+
+const std::map<Token, std::function<std::string()>> moduleFunctions = {
```
Comment (cieslarmichal) [inline include/faker-cxx/Structure.h:348]: move this mapping to cpp file into anonymouse namespace please

File: include/faker-cxx/Structure.h [hunk 4]
```diff
@@ -0,0 +1,662 @@
...
+    {Token::WordAdjective, []() { return Word::adjective(); }},
+    {Token::WordAdverb, []() { return Word::adverb(); }},
+    {Token::WordConjuction, []() { return Word::conjunction(); }},
+    {Token::WordInterjection, []() { return Word::interjection(); }},
+    {Token::WordNoun, []() { return Word::noun(); }},
+    {Token::WordPreposition, []() { return Word::preposition(); }},
+    {Token::WordVerb, []() { return Word::verb(); }},
+};
+
+class Sructure
```
Comment (cieslarmichal) [inline include/faker-cxx/Structure.h:633]: typo

File: include/faker-cxx/Structure.h [hunk 5]
```diff
@@ -0,0 +1,662 @@
...
+{
+public:
+    /**
+     * @brief Get a json string based on input 
+     *
+     * @return a josn string based on user selected tokens 
+     *
+     * @code
+     * Sructure::json(tokens) // {"FoodName":"Cajun Shrimp Pasta","airlineType":"narrowbody","airportName":"Brisbane International Airport"}
```
Comment (cieslarmichal) [inline include/faker-cxx/Structure.h:642]: please add creating example tokens into example

File: src/common/JsonHelper.h [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+
+namespace faker
+{
+
+class JsonHelper
+{
+public:
+// this function parse simple jason like {"key1":"value1","key2":"value2"}
+    static std::map<std::string, std::string> simpleJsonParser(std::string json);
+
```
Comment (cieslarmichal) [inline src/common/JsonHelper.h:15]: remove redundant new lines

**Review by cieslarmichal — CHANGES_REQUESTED** [review]
Good job, couple of things to change :)

File: .gitmodules (hunk 1 above)
Comment (alisedighmoghadam) [reply .gitmodules]: I don't know how this got there but I fixed it.

**Review by cieslarmichal — APPROVED** [review]

**Comment by cieslarmichal** [comment]
Nice, good job :)

**Pull request merged**